libbeat/monitoring/monitoring.go

```diff
@@ -17,7 +17,19 @@
 
 package monitoring
 
-import "errors"
+import (
+	"errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/monitoring/report"
+)
+
+// MonitoringBeatConfig represents the part of the $BEAT.yml to do with monitoring settings
+type MonitoringBeatConfig struct {
```

Comment:
I originally tried to name this just `BeatConfig` to avoid the naming stutter. However, that didn't quite work out with composing this struct into the [`instance.beatConfig` struct](https://github.com/elastic/beats/pull/9260/files#diff-f7226280e7734d6e3d88e7dde232602eR87), since the latter already composes in the [`beat.BeatConfig` struct](https://github.com/elastic/beats/pull/9260/files#diff-f7226280e7734d6e3d88e7dde232602eL88), thereby resulting in a naming collision.

Reply:
The naming conflict is due to:

```
	// monitoring settings
	monitoring.MonitoringBeatConfig `config:",inline"`
```

One could have resolved it via:

```
	// monitoring settings
	MonitoringBeatConfig monitoring.BeatConfig `config:",inline"`
```

When embedding a struct you create an anonymous field that is accessed via the struct it's name.

Reply:
Addressed in 4573a30.